submissions/Ant-C-tech/friends-app/js/app.js

```diff
@@ -0,0 +1,314 @@
+
+    MAIN.classList.remove('scroll')
+    if (speed) {
+        MAIN.classList.add(speed)
+        CURRENT_ANIMATION_SPEED = speed
+    }
+    MAIN.classList.add(hide)
+}
+
+function toSearch() {
+    const input = SEARCH_INPUT.value.toLowerCase()
+
+    let foundFriends = (CURRENT_FRIENDS.length === 0) ?
+        FRIENDS_SOURCE.filter(friend => (friend.name.first.toLowerCase().startsWith(input) || friend.name.last.toLowerCase().startsWith(input))) :
+        CURRENT_FRIENDS.filter(friend => (friend.name.first.toLowerCase().startsWith(input) || friend.name.last.toLowerCase().startsWith(input)))
+
+    changeContent(createFriendsScreen(foundFriends), SHOW_ELEM_SECONDARY_ANIMATION, HIDE_ELEM_SECONDARY_ANIMATION, ANIMATION_SPEED)
+}
+
+function toFilter() {
+    //In case user enter to app through sidenav (not using startButton)
+    if (!IS_MUSIC_STOPPED_BY_USER) {
+        playMusic()
+    }
+    showSearchBar()
+    closeSideNav()
+
+    const checkedGenderInput = document.querySelector("input[type='radio'][name='gender']:checked")
+    let userChooseGender = (checkedGenderInput) ? [checkedGenderInput.getAttribute("data-gender")] : ['male', 'female']
+
+    const [userChooseMinAge, userChooseMaxAge] = document.getElementById('test-slider').noUiSlider.get()
+
+    //Filter
+    CURRENT_FRIENDS = FRIENDS_SOURCE.filter(friend => (
+        userChooseGender.includes(friend.gender) &&
+        friend.dob.age >= userChooseMinAge &&
+        friend.dob.age <= userChooseMaxAge))
+
+    const checkedSortParameterInput = document.querySelector("input[type='radio'][name='sort']:checked")
+    let sortParameter = (checkedSortParameterInput) ? checkedSortParameterInput.getAttribute("data-sort") : false
+
+    //Sort
+    switch (sortParameter) {
+        case '0-100':
+            CURRENT_FRIENDS.sort(byField('dob', 'age'))
+            break
+        case '100-0':
+            CURRENT_FRIENDS.sort(byField('dob', 'age')).reverse()
+            break
+        case 'name_a-z':
+            CURRENT_FRIENDS.sort(byField('name', 'first'))
+            break
+        case 'name_z-a':
+            CURRENT_FRIENDS.sort(byField('name', 'first')).reverse()
+            break
+        case 'lastName_a-z':
+            CURRENT_FRIENDS.sort(byField('name', 'last'))
+            break
+        case 'lastName_z-a':
+            CURRENT_FRIENDS.sort(byField('name', 'last')).reverse()
+            break
+
+        default:
+            break
+    }
+
+    changeContent(createFriendsScreen(CURRENT_FRIENDS), SHOW_ELEM_PRIMARY_ANIMATION, HIDE_ELEM_PRIMARY_ANIMATION)
+}
+
+function resetFilter() {
+    //In case user enter to app through sidenav (not using startButton)
+    if (!IS_MUSIC_STOPPED_BY_USER) {
+        playMusic()
+    }
+    showSearchBar()
+    closeSideNav()
+
+    CURRENT_FRIENDS = []
+    if (document.querySelector("input[type='radio'][name='gender']:checked")) {
+        document.querySelector("input[type='radio'][name='gender']:checked").checked = false
```

Comment:
Searching for element in the DOM tree takes time and resources. Also, it is a long line which hard to read.

So I suggest you use a variable with results of the query selector instead.